tests/tests/mempool.rs

```diff
@@ -0,0 +1,396 @@
+
+    // contains function test
+    assert!(contains);
+}
+
+#[rstest]
+#[awt]
+#[tokio::test(flavor = "multi_thread")]
+async fn test_mempool_add_external_transactions(#[future] katana: Katana, _setup: ()) {
+    let eth_provider = katana.eth_provider();
+
+    // Create multiple sample transactions
+    let transactions = create_sample_transactions(&katana, 2).await.expect("Failed to create sample transaction");
+
+    // Collect pooled transactions
+    let pooled_transactions =
+        transactions.iter().map(|(eth_pooled_transaction, _)| eth_pooled_transaction.clone()).collect::<Vec<_>>();
+
+    // Collect signed transactions
+    let signed_transactions =
+        transactions.iter().map(|(_, signed_transactions)| signed_transactions.clone()).collect::<Vec<_>>();
+
+    // Add external transactions to mempool
+    let _ = eth_provider.mempool().unwrap().add_external_transactions(pooled_transactions).await;
+
+    // Get pooled transaction hashes
+    let hashes = eth_provider.mempool().unwrap().pooled_transaction_hashes();
+
+    // Set maximum number of hashes to retrieve
+    let hashes_max_number = 1;
+
+    // Get pooled transaction hashes with a limit
+    let hashes_max = eth_provider.mempool().unwrap().pooled_transaction_hashes_max(hashes_max_number);
+
+    // Get external transactions
+    let external_transactions = eth_provider.mempool().unwrap().get_external_transactions();
+
+    // Get updated mempool size
+    let mempool_size = eth_provider.mempool().unwrap().pool_size();
+    // Check pending transactions
+    assert_eq!(mempool_size.pending, 2);
+    // Check queued transactions
+    assert_eq!(mempool_size.queued, 0);
+    // Check total transactions
+    assert_eq!(mempool_size.total, 2);
+
+    // pooled_transaction_hashes function tests
+    // Check if the first signed transaction hash is present
+    assert!(hashes.contains(&signed_transactions[0].hash()));
+    // Check if the second signed transaction hash is present
+    assert!(hashes.contains(&signed_transactions[1].hash()));
+    // Ensure the hashes are not empty
+    assert!(!hashes.is_empty());
+
+    // pooled_transaction_hashes_max function test
+    // Check if at least one signed transaction hash is present
+    assert!(hashes_max.contains(&signed_transactions[0].hash()) || hashes_max.contains(&signed_transactions[1].hash()));
+    // Ensure the number of hashes matches the limit
+    assert_eq!(hashes_max.len(), hashes_max_number);
+    // Ensure the hashes are not empty
+    assert!(!hashes_max.is_empty());
+
+    // get_external_transactions function test
+    // Check if the returned transaction hash matches
+    assert_eq!(*external_transactions[0].hash(), signed_transactions[0].hash());
+    assert_eq!(*external_transactions[1].hash(), signed_transactions[1].hash());
+}
+
+#[rstest]
+#[awt]
+#[tokio::test(flavor = "multi_thread")]
+async fn test_mempool_add_transaction_and_subscribe(#[future] katana: Katana, _setup: ()) {
```

Comment:
we aren't really testing the subscription here right?

Reply:
No, it wasn’t testing the subscription. I removed it, I don’t think it was useful. What do you think?

Reply:
I think it makes sense to remove if you weren't actually testing the subscription yes